src/post/post-interaction-counts.service.ts

```diff
@@ -0,0 +1,98 @@
+import type { Logger } from '@logtape/logtape';
+import { getError, isError } from 'core/result';
+import type { KnexPostRepository } from './post.repository.knex';
+import type { PostService } from './post.service';
+
+export class PostInteractionCountsService {
+    constructor(
+        private readonly postService: PostService,
+        private readonly postRepository: KnexPostRepository,
+        private logger: Logger,
+    ) {}
+
+    /**
+     * Updates the interaction counts for the given post IDs, if the update is due.
+     *
+     * @param {number[]} postIds - The IDs of the posts to update
+     */
+    async updateInteractionCounts(postIds: number[]) {
+        for (const postId of postIds) {
+            const post = await this.postRepository.getById(postId);
+
+            if (!post) {
+                this.logger.error(
+                    'Post with ID {postId} not found when updating interaction counts - Skipping',
+                    { postId },
+                );
+                continue;
+            }
+
+            if (!this.isUpdateDue(post.publishedAt, post.updatedAt)) {
+                this.logger.info(
+                    'Post with ID {postId} is not due for an update of interaction counts - Skipping',
+                    { postId },
+                );
+                continue;
+            }
+
+            const result = await this.postService.updateInteractionCounts(post);
+
+            if (isError(result)) {
+                this.logger.error(
+                    'Error updating interaction counts for post with ID {postId}: {error}',
+                    { postId, error: getError(result) },
+                );
+                continue;
+            }
+
+            this.logger.info(
+                'Successfully updated interaction counts for post with ID {postId}',
+                { postId },
+            );
+        }
+    }
+
+    /**
+     * Computes whether the post is due an update of like/repost counts.
+     *
+     * @table
+     * | Post published       | Refresh interaction counts       |
+     * |----------------------|----------------------------------|
+     * | < 6 hours ago        | At most once every 10 minutes    |
+     * | 6–24 hours ago       | At most once every 2 hours       |
+     * | 1-7 days ago         | At most once every 6 hours       |
+     * | > 7 days ago         | At most once per day             |
+     *
```

Comment:
Nice, although is `@table` a valid JSDoc thing? (either way the table is still useful)

Reply:
Ah good one - I've seen it recently in some code but it doesn't seem to be part of JSDoc core, removed!